cpp/src/arrow/io/file.cc

```diff
@@ -152,20 +152,39 @@ static inline int64_t lseek64_compat(int fd, int64_t pos, int whence) {
 #endif
 }
 
+#if defined(_MSC_VER)
+static inline Status Utf8ToUtf16(const std::string& input, std::wstring &result) {
```

Comment:
Is it possible to use the C++11 codecvt header for this? e.g. https://github.com/wesm/feather/blob/master/cpp/src/feather/io.cc#L169